From 62779bcb89c9e84d5dc4f5b274c9357c773f94ec Mon Sep 17 00:00:00 2001
From: Sandy Huang <hjc@rock-chips.com>
Date: Wed, 31 Jan 2018 10:20:48 +0800
Subject: [PATCH] drm/rockchip: backlight: fix compile warning

warning: Access to field 'dev' results in a dereference of a null
pointer (loaded from variable 'panel')

Change-Id: I0f4224b47b3b7c16a9022e19f8de28df66784a7f
Signed-off-by: Sandy Huang <hjc@rock-chips.com>
---
 drivers/gpu/drm/rockchip/rockchip_drm_backlight.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/rockchip/rockchip_drm_backlight.c b/drivers/gpu/drm/rockchip/rockchip_drm_backlight.c
index 21a743052cb0..52c761f79978 100644
--- a/drivers/gpu/drm/rockchip/rockchip_drm_backlight.c
+++ b/drivers/gpu/drm/rockchip/rockchip_drm_backlight.c
@@ -307,7 +307,7 @@ static int rockchip_drm_backlight_bind(struct device *dev,
 
 	drm_for_each_connector(connector, drm_dev) {
 		panel = drm_find_panel_by_connector(connector);
-		if (!panel && !panel->dev)
+		if (!panel || !panel->dev)
 			continue;
 		backlight_np = of_parse_phandle(panel->dev->of_node,
 						"backlight", 0);
-- 
2.35.3

